packages/database/src/core/SyncTree.ts

```diff
@@ -347,72 +350,76 @@ export function syncTreeRemoveEventRegistration(
     if (syncPointIsEmpty(maybeSyncPoint)) {
       syncTree.syncPointTree_ = syncTree.syncPointTree_.remove(path);
     }
+
     const removed = removedAndEvents.removed;
     cancelEvents = removedAndEvents.events;
-    // We may have just removed one of many listeners and can short-circuit this whole process
-    // We may also not have removed a default listener, in which case all of the descendant listeners should already be
-    // properly set up.
-    //
-    // Since indexed queries can shadow if they don't have other query constraints, check for loadsAllData(), instead of
-    // queryId === 'default'
-    const removingDefault =
-      -1 !==
-      removed.findIndex(query => {
-        return query._queryParams.loadsAllData();
-      });
-    const covered = syncTree.syncPointTree_.findOnPath(
-      path,
-      (relativePath, parentSyncPoint) =>
-        syncPointHasCompleteView(parentSyncPoint)
-    );
 
-    if (removingDefault && !covered) {
-      const subtree = syncTree.syncPointTree_.subtree(path);
-      // There are potentially child listeners. Determine what if any listens we need to send before executing the
-      // removal
-      if (!subtree.isEmpty()) {
-        // We need to fold over our subtree and collect the listeners to send
-        const newViews = syncTreeCollectDistinctViewsForSubTree_(subtree);
+    if (!skipListenerDedup) {
+      // We may have just removed one of many listeners and can short-circuit this whole process
+      // We may also not have removed a default listener, in which case all of the descendant listeners should already be
+      // properly set up.
+      //
+      // Since indexed queries can shadow if they don't have other query constraints, check for loadsAllData(), instead of
+      // queryId === 'default'
```

Comment:
Let's use a consistent comment style.

Reply:
For a two-lined comment, a block comment seems too much. I'll update the comment above though.